teuthology/openstack/__init__.py

```diff
@@ -1159,6 +1163,13 @@ def create_security_group(self):
 openstack security group rule create --src-group {server} --dst-port 1:65535 {server}
 openstack security group rule create --protocol udp --src-group {server} --dst-port 1:65535 {server}
         """.format(server=self.server_group(), worker=self.worker_group()))
+        if self.args.nameserver:
+            misc.sh("""
+openstack security group rule create --protocol tcp --src-ip {nameserver} --dst-port 53 {server} # for DNS access
+openstack security group rule create --protocol udp --src-ip {nameserver} --dst-port 53 {server} # for DNS access
+openstack security group rule create --protocol tcp --src-ip {nameserver} --dst-port 53 {worker} # for DNS access
+openstack security group rule create --protocol udp --src-ip {nameserver} --dst-port 53 {worker} # for DNS access
```

Comment:
I would think it's enough to give the target nodes access to port 53 of the nameserver. Can't see a scenario where the nameserver will try to connect to target nodes.